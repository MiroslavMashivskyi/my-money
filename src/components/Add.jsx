import React from 'react';
import { reduxForm, Field } from 'redux-form';
import Currency_Container from './Currency_Container';
import require from '../utils/validator';
import { Input, Select } from './Input';

const Add = (props) => {
  const onSubmit = ({ date, amount, currency, goodsName, reset }) => {
    props.addNote(date, amount, currency, goodsName);
    props.add(amount, currency);
    props.reset('add');
    props.clear();
  }
  
  return (
    <AddReduxForm onSubmit={onSubmit}/>
  );
}

const AddForm = (props) => {
  return (
    <form className="form add" onSubmit={props.handleSubmit}>
      <button className = "button">Add</button>
      <Field component={Input} name="date" type="date" validate={require}/>
      <Field component={Input} name="amount" type="number" validate={require}/>
      <Field component={Select} name="currency" validate={require}>
        <option></option>
        <Currency_Container />
      </Field>
      <Field component={Input} name="goodsName" type="text" validate={require}/>
    </form>
  )
}

const AddReduxForm = reduxForm({form: "add"})(AddForm);
export default Add;