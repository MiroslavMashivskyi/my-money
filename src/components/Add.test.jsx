import React from 'react';
import { shallow } from 'enzyme';
import Add from './Add';

test("Add should work correctly", () => {
  expect(shallow(<Add />).debug()).toMatchSnapshot();
});