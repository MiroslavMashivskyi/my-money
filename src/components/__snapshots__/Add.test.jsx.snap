// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add should work correctly 1`] = `"<ReduxForm onSubmit={[Function: onSubmit]} touchOnBlur={true} touchOnChange={false} persistentSubmitErrors={false} destroyOnUnmount={true} shouldAsyncValidate={[Function: defaultShouldAsyncValidate]} shouldValidate={[Function: defaultShouldValidate]} shouldError={[Function: defaultShouldError]} shouldWarn={[Function: defaultShouldWarn]} enableReinitialize={false} keepDirtyOnReinitialize={false} updateUnregisteredFields={false} getFormState={[Function: getFormState]} pure={true} forceUnregisterOnUnmount={false} submitAsSideEffect={false} form=\\"add\\" />"`;
